### ISSUE TITLE
Consider deprecating Flapjack

### ISSUE THREAD

**Issue by ptomato** [endlessm/flapjack#29]
It may have outlived its usefulness now that the freedesktop and GNOME runtimes are built with Buildstream.

**Comment by bam80** [comment]
I didn't use Flapjack but does it only useful for SDKs? What if I want to hack on one of the bundled dependencies in my flatpak?

**Comment by ptomato** [comment]
It is intended for SDKs. For apps there was already https://gitlab.com/csoriano/flatpak-dev-cli before flapjack existed.

**Comment by bam80** [comment]
Thanks. No evidence if it's still supported, though: https://gitlab.com/csoriano/flatpak-dev-cli/-/issues/9

**Comment by dylanmccall** [comment]
Yep, I agree. I was working on some tools bolted on to BuildStream that help us achieve a Flapjack-like workflow with the Endless SDK by using _workspaces_ and some other little tricks. I wrote some notes over here with some hacky scripts for reference:

https://gist.github.com/dylanmccall/fa3e108ee35a766f8b8ebba0073c8026

It's stalled at the moment, but I'd love to get back to that, sync up with where BuildStream is going currently, and put out a proper "here's what you can do now" kind of document.

Speaking of workflows, I should mention that I added tests to the Endless SDK's buildstream project, so instead of running tests with `flapjack test`, we build the _test_ element for each component our CI system cares about: https://github.com/endlessm/endless-sdk-flatpak/tree/master/elements/tests. It was fiddly at first since there's a bunch of environment set-up to be done (unearthing lots of those issues that are not really bugs unless we're being incredibly pedantic), but it works _spectacularly_ well once it works, so I highly recommend it :)

**Comment by ptomato** [comment]
Oh, nice! That was one of the things I tried to do with BuildStream when I originally wrote Flapjack. But I failed at building xvfb-run with BuildStream; or rather I succeeded at building it but the built binary would crash randomly when running in the BuildStream sandbox. I never figured it out, but I'm glad to see that you did!
